cmd/trtl/main.go

```diff
@@ -530,6 +546,195 @@ func initPeersClient(c *cli.Context) (err error) {
 // Trtl (DB) Client Functions
 //===========================================================================
 
+// migrateCerts creates certificate records in the trtl database based on the existing certreqs.
+func migrateCerts(c *cli.Context) (err error) {
+	ctx, cancel := profile.Context()
+	defer cancel()
+
+	// Open the report file if requested
+	var report *os.File
+	if c.String("report") != "" {
+		if report, err = os.Create(c.String("report")); err != nil {
+			return cli.Exit(fmt.Errorf("could not open report file %s: %s", c.String("report"), err), 1)
+		}
+		defer report.Close()
+	}
+
+	// Get the list of certreqs
+	req := &pb.CursorRequest{
+		Namespace: wire.NamespaceVASPs,
+	}
+	var stream pb.Trtl_CursorClient
+	if stream, err = dbClient.Cursor(ctx, req); err != nil {
+		return cli.Exit(err, 1)
+	}
+
+	var migrated, skipped int
+	var migrateErr error
+
+	for {
+		var rep *pb.KVPair
+		if rep, err = stream.Recv(); err != nil {
+			if err != io.EOF {
+				migrateErr = err
+			}
+			break
+		}
+
+		vasp := &gds.VASP{}
+		if err = proto.Unmarshal(rep.Value, vasp); err != nil {
+			migrateErr = err
+			break
+		}
+
+		// Get all the certificate requests for this vasp
+		var ids []string
+		if ids, err = models.GetCertReqIDs(vasp); err != nil {
+			migrateErr = err
+			break
+		}
+
+		// Get the latest completed certificate request
+		var certreq *models.CertificateRequest
+		for _, id := range ids {
+			var rep *pb.GetReply
+			if rep, err = dbClient.Get(ctx, &pb.GetRequest{
+				Namespace: wire.NamespaceCertReqs,
+				Key:       []byte(id),
+			}); err != nil {
+				fmt.Printf("WARNING: could not get certreq %s for vasp %s (%s): %s\n", id, vasp.Id, vasp.CommonName, err)
+				if report != nil {
+					report.WriteString(fmt.Sprintf("WARNING: could not get certreq %s for vasp %s (%s): %s\n", id, vasp.Id, vasp.CommonName, err))
+				}
+				continue
+			}
+
+			current := &models.CertificateRequest{}
+			if err = proto.Unmarshal(rep.Value, current); err != nil {
+				migrateErr = err
+				break
+			}
+
+			if current.Status == models.CertificateRequestState_COMPLETED {
+				certreq = current
+			}
```

Comment:
so I'm not sure this logic is 100% correct - a vasp might have multiple completed certificate requests; this does not imply that it is the latest request without timestamp checks. 

Multiple completed certreqs will often happen on reissuance. 

Also - do we want to create a certificate for all completed cert reqs and then mark the ones that need to be updated with certificate info? I know that might be a manual process.

Reply:
So should we check the `Created` timestamp to determine the newest completed certificate to put the identity certificate in and create temporary records for the completed ones? It's tricky because we don't know the serial number to create the ID with but maybe we can populate those with a random UUID and when we do the manual cleanup, we already know which ones need to be corrected.

Reply:
The Created timestamp methodology works for me. 

Instead of creating temporary records, would it be easier to mark them in the report (e.g. via certificate request id), and then we can create the records manually later?